packages/core/contracts/financial-templates/perpetual-multiparty/PositionManager.sol

```diff
@@ -101,37 +99,29 @@ contract PositionManager is FeePayer, AdministrateeInterface {
     event NewSponsor(address indexed sponsor);
     event EndedSponsorPosition(address indexed sponsor);
     event Redeem(address indexed sponsor, uint256 indexed collateralAmount, uint256 indexed tokenAmount);
-    event ContractExpired(address indexed caller);
-    event SettleExpiredPosition(
+    event EmergencyShutdown(address indexed caller, uint256 shutdownTimestamp);
+    event SettleEmergencyShutdown(
         address indexed caller,
         uint256 indexed collateralReturned,
         uint256 indexed tokensBurned
     );
-    event EmergencyShutdown(address indexed caller, uint256 originalExpirationTimestamp, uint256 shutdownTimestamp);
 
     /****************************************
      *               MODIFIERS              *
      ****************************************/
 
-    modifier onlyPreExpiration() {
-        _onlyPreExpiration();
+    modifier onlyCollateralizedPosition(address sponsor) {
+        _onlyCollateralizedPosition(sponsor);
         _;
     }
 
-    modifier onlyPostExpiration() {
-        _onlyPostExpiration();
+    modifier notEmergencyShutdown() {
+        _notEmergencyShutdown();
         _;
     }
 
-    modifier onlyCollateralizedPosition(address sponsor) {
-        _onlyCollateralizedPosition(sponsor);
-        _;
-    }
-
-    // Check that the current state of the pricelessPositionManager is Open.
-    // This prevents multiple calls to `expire` and `EmergencyShutdown` post expiration.
-    modifier onlyOpenState() {
-        _onlyOpenState();
+    modifier onlyEmergencyShutdown() {
```

Comment:
Why have both `notEmergencyShutdown()` and `onlyEmergencyShutdown`?

Reply:
Some logic should be only executed iff the contract has not emergency shutdown (basically all existing functionality). The `settleEmergencyShutdown` should however only be called if the contract has been EmergencyShutdown. Are you saying rather put this logic in the `settleEmergencyShutdown` function and then have 1 modifier?

Reply:
I think alternatively, having 2 modifiers and just 1 internal method might be better